dataprovider/providers/aws/aws_data_provider.go

```diff
@@ -0,0 +1,77 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package aws
+
+import (
+	"context"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/cloudbeat/dataprovider/types"
+	"github.com/elastic/cloudbeat/version"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const (
+	cloudAccountIdField   = "cloud.account.id"
+	cloudAccountNameField = "cloud.account.name"
+	cloudProviderField    = "cloud.provider"
+	cloudProviderValue    = "aws"
+)
+
+type AWSDataProvider struct {
+	log         *logp.Logger
+	accountId   string
+	accountName string
+}
+
+func New(options ...Option) AWSDataProvider {
+	adp := AWSDataProvider{}
+	for _, opt := range options {
+		opt(&adp)
+	}
+	return adp
+}
+
+func (a AWSDataProvider) FetchData(ctx context.Context, resource string, id string) (types.Data, error) {
```

Comment:
I'd replace it with a [blank identifier](https://go.dev/ref/spec#Blank_identifier) for unused params

```suggestion
func (a AWSDataProvider) FetchData(_ context.Context, _ string, id string) (types.Data, error) {
```